app/src/main/java/com/kickstarter/viewmodels/MessagesViewModel.java

```diff
@@ -75,129 +103,211 @@ public ViewModel(final @NonNull Environment environment) {
       this.client = environment.apiClient();
       this.currentUser = environment.currentUser();
 
-      final Observable<MessageThread> messageThread = intent()
-        .take(1)
-        .map(i -> i.getParcelableExtra(IntentKey.MESSAGE_THREAD))
-        .ofType(MessageThread.class);
+      final Observable<Either<MessageThread, Pair<Project, Backing>>> configData = intent()
+        .map(i -> {
+          final MessageThread messageThread = i.getParcelableExtra(IntentKey.MESSAGE_THREAD);
+          return messageThread != null
+            ? new Either.Left<MessageThread, Pair<Project, Backing>>(messageThread)
+            : new Either.Right<MessageThread, Pair<Project, Backing>>(
+              Pair.create(i.getParcelableExtra(IntentKey.PROJECT), i.getParcelableExtra(IntentKey.BACKING))
+          );
+        });
+
+      final Observable<KoalaContext.Message> koalaContext = intent()
+        .map(i -> i.getSerializableExtra(IntentKey.KOALA_CONTEXT))
+        .ofType(KoalaContext.Message.class);
+
+      final Observable<Backing> configBacking = configData
+        .map(Either::right)
+        .filter(ObjectUtils::isNotNull)
```

Comment:
ahh, but for places where we want to keep point free notation, we can't also use a static import